org.envirocar.app/src/org/envirocar/app/views/obdselection/OBDSelectionFragment.java

```diff
@@ -115,8 +130,9 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
         // Setup the paired devices.
         updatePairedDevicesList();
 
-        // Start the discovery of bluetooth devices.
-        updateContentView();
```

Comment:
Thank for reviewing this Arvind. With `updatecontentview()` in `OnCreate` GPS request dialog appears 2 times, because `updatecontentview()`  is called 2 times. Once in `onCreateView` and next time in `onBluetoothStateChangedEvent` . I have replace this so that the `requestGps()` method is called only 1 time. It would be helpful if you could suggest any alternate approach so that `requestGps` method is called only once.

Reply:
https://user-images.githubusercontent.com/70392921/123386066-e30bdd80-d5b3-11eb-9725-14bac67e9f80.mp4